.github/actions/artifact_upload/action.yml

```
@@ -0,0 +1,50 @@
+name: Upload artifact
+description: Upload an encrypted zip archive as a github artifact.
+inputs:
+  path:
+    description: 'The path(s) that should be uploaded.'
+    required: true
+  name:
+    description: 'The name of the artifact.'
+    required: true
+  retention-days:
+    description: 'How the artifact should be retained.'
+    required: false
+    default: 60
+  encryption-secret:
+    description: 'The secret to use for encrypting the files.'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Install zip
+      uses: montudor/action-zip@v1
+    - name: Find glob pattern matches
+      uses: tj-actions/glob@v17
+      id: glob
+      with:
+        files: |
+          ${{ inputs.path }}
+    - name: Create temporary directory
+      id: tempdir
+      shell: bash
+      run: echo "directory=$(mktemp -d)" >> "$GITHUB_OUTPUT"
+    - name: Create archive
+      shell: bash
+      run: |
+        # Create an archive if files exist.
+        # Don't create an archive file if no files are found
+        # and warn/error if configured that way.
+        if [ -n "${{ steps.glob.outputs.paths }}" ]
```

```suggestion
        if [[ -n "${{ steps.glob.outputs.paths }}" ]]
```
we use [[ per convention.